packages/@aws-cdk/cdk/lib/synthesis.ts

```diff
@@ -74,14 +73,16 @@ export class SynthesisSession implements ISynthesisSession {
     return 'synthesize' in obj;
   }
 
-  public readonly store: ISessionStore;
+  public readonly outdir: string;
 
   private readonly artifacts: { [id: string]: cxapi.Artifact } = { };
-  private readonly buildSteps: { [id: string]: cxapi.BuildStep } = { };
   private _manifest?: cxapi.AssemblyManifest;
 
-  constructor(options: SynthesisOptions) {
-    this.store = options.store || new InMemoryStore();
+  constructor(options: SynthesisOptions = { }) {
+    this.outdir = options.outdir || fs.mkdtempSync(path.join(os.tmpdir(), 'cdk.out'));
+    if (!fs.existsSync(this.outdir)) {
+      fs.mkdirSync(this.outdir);
```

Comment:
That'll create only a leaf, won't it? Should do `mkdir -p` style if you're going to create for the user, I think.

Reply:
The node standard library doesn't have an equivalent of `mkdir -p`, and I wanted to avoid the dependency. This is pragmatic.